### DISCUSSION TITLE
I cannot get SSL working for the server

### DISCUSSION THREAD

**Discussion by tre4b** [FreshRSS/FreshRSS#5591] [Q&A]
Has anyone got a guide on this?  I suspect it is because I am using a self certified certificate and I can't work out how to add this to the FreshRSS servers CA list. 

I am running FreshRSS via TrueNas/TrueCharts and using an instance of HA to get to the server at the given port without having to type it in.  All other servers I do this on work fine, this one though refuses to work.   Ironically the certificate it shows in the error in my browser is the correct certificate, it is just not considered to be the correct certificate, probably because of the HA cert offloading.  If I force SSL between HA and the server then I just get a 503 error.  I can only connect if the connection from the HA to the server is open.

![cert problem](https://github.com/FreshRSS/FreshRSS/assets/37894577/e72b4f8c-1324-45e2-94cf-fccfe6f54bd1)

**Comment by Frenzie** [comment]
While I don't know if I can help you, I don't think anyone can if you don't provide some basic troubleshooting info. ;-) See in the issue template:

```
**Environment information (please complete the following information):**
- Device: [e.g. iPhone6]
- OS: [e.g. Ubuntu 20.04]
- Browser: [e.g. Firefox 86]
- FreshRSS version: [e.g. 1.17.1]
- Database version: [e.g. Mysql 5.7]
- PHP version: [e.g. PHP 7.4]
- Installation type: [e.g. Yunohost]
```

**Comment by tre4b** [comment]
Thanks Frenzie,

I am not sure either,  my challenge is that I installed this as a container in Truenas.   I am using the most up to date container but I don't know what underlying OS/FreshRSS/Database/PHP version etc.  It is possible too that the container is the issue.  At the moment I am trying to get to the bottom of this stuff.  I was really just seeking whether there was information on how FreshRSS stores its certificates.

I have found that the container has openssl installed so am currently assuming that may help me.  Probably best if I keep playing and come back if I have a more specific question I can ask.

**Reply by Frenzie** [reply]
>  I was really just seeking whether there was information on how FreshRSS stores its certificates.

It doesn't, basically? When using containers you'd normally use some kind of reverse proxy like nginx or Traefik. It's only if you use nginx or Apache directly that you wouldn't.

**Reply by tre4b** [reply]
Makes sense,  it has nginx installed in the container.  Problems are probably compounded by me using HAProxy to connect to the servers too.   Feels like a problem of my own making sadly.  I'm happily using freshrss in a browser,  I just can't attach an app or anything as they don't like the certificate situation.

**Reply by Alkarex** [reply]
As Frenzie indicates, our official Docker images just have a basic HTTP Web server (Apache on Linux - Debian or Alpine), and it is expected that you put in front some kind of reverse proxy handling HTTPS termination and other aspects.
See for instance our Docker production guide https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#run-in-production
